src/math/minkowski_difference.rs

```diff
@@ -0,0 +1,86 @@
+/// The Minkowski distance or Minkowski metric is a metric in a normed vector space which
+/// can be considered as a generalization of both the Euclidean distance and the Manhattan distance.
+/// It is named after the Polish mathematician Hermann Minkowski.
+/// https://en.wikipedia.org/wiki/Minkowski_distance
+///
+/// This metric is useful in the field of machine learning to judge the similarity between data points.
+/// It's also employed for collision detection in game engines.
+use num_traits::Pow;
+
+pub struct Point {
+    pub x: Vec<f64>,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum ParameterError {
+    DifferentDimensions,
+    InvalidOrder,
+}
+
+pub fn minkowski_difference(a: &Point, b: &Point, order: u32) -> Result<f64, ParameterError> {
```

Comment:
```suggestion
pub fn minkowski_distance(a: &Point, b: &Point, order: f64) -> Result<f64, ParameterError> {
```
There are two things:
- the name: why not `minkowski_distance`?
- why to limit `order` to `u32`?